algorithms/src/std_algorithms/impl/Kokkos_Constraints.hpp

```diff
@@ -215,6 +230,37 @@ KOKKOS_INLINE_FUNCTION void expect_valid_range(IteratorType first,
   (void)last;
 }
 
+//
+// Check if kokkos iterators are overlapping
+//
+template <typename IteratorType1, typename IteratorType2>
+KOKKOS_INLINE_FUNCTION void expect_no_overlap(
+    [[maybe_unused]] IteratorType1 first, [[maybe_unused]] IteratorType1 last,
+    [[maybe_unused]] IteratorType2 s_first,
+    [[maybe_unused]] IteratorType2 s_last) {
+  if constexpr (is_kokkos_iterator_v<IteratorType1> &&
+                is_kokkos_iterator_v<IteratorType2>) {
+    if constexpr (std::is_constructible_v<IteratorType2, IteratorType1>) {
+      IteratorType2 first2(first), last2(last);
+      IteratorType2 next_first2           = first2;
+      ptrdiff_t stride                    = &*(++next_first2) - &*first2;
```

Comment:
Instead, we could avoid executing this in the correct space if we allow to retrieve the View from the iterator.